client/src/components/ui/Modal.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+    import Button from "./Button.svelte";
+
+    export let showModal = false;
+    export let isForm = false;
+
+    let dialog: HTMLDialogElement;
+
+    $: if (dialog && showModal) dialog.showModal();
```

Comment:
We should only check for the `dialog` once we know we must `showModal`.
```suggestion
    $: if (showModal) dialog?.showModal();
```